myriad-scheduler/src/main/java/com/ebay/myriad/state/SchedulerState.java

```diff
@@ -191,7 +192,21 @@ public synchronized void removeTask(Protos.TaskID taskId) {
         return Collections.unmodifiableCollection(activeNodeTasks);
     }
 
-    // TODO (sdaingade) Clone NodeTask
+    public synchronized Collection<NodeTask> getActiveTasksForProfile(NMProfile profile) {
+      List<NodeTask> activeNodeTasks = new ArrayList<>();
+      if (CollectionUtils.isNotEmpty(activeTasks)
+          && CollectionUtils.isNotEmpty(tasks.values())) {
+        for (Map.Entry<Protos.TaskID, NodeTask> entry : tasks.entrySet()) {
+          NodeTask nodeTask = entry.getValue();
+          if (activeTasks.contains(entry.getKey()) && nodeTask.getProfile().getName().equals(profile.getName())) {
+            activeNodeTasks.add(nodeTask);
+          }
+        }
+      }
+      return Collections.unmodifiableCollection(activeNodeTasks);
```

Comment:
Not sure why. I just made a new method from ```getActiveTasks()``` that takes ```NMProfile``` as parameter and filters the active tasks by the profile.
@sdaingade - do you know why we need an unmodifiable collection for all the returned objects of ```SchedulerState``` ?.